tests/codegen/vec_pop_push_noop.rs

```diff
@@ -9,7 +9,7 @@ pub fn noop(v: &mut Vec<u8>) {
     // CHECK-NOT: call
     // CHECK: tail call void @llvm.assume
     // CHECK-NOT: grow_one
-    // CHECK-NOT: call
+    // CHECK-NOT: call {{(v(o(i(d( (@(l(l(v(m(\.(a(s(s(u(m([^e]|$)|[^m]|$)|[^u]|$)|[^s]|$)|[^s]|$)|[^a]|$)|[^.]|$)|[^m]|$)|[^v]|$)|[^l]|$)|[^l]|$)|[^@]|$)|[^ ]|$)|[^d]|$)|[^i]|$)|[^o]|$)|[^v]|$)}}
```

Comment:
Congratulations, you won the unreadable codegen test contest! 😜

Reply:
If the regular expression engine used by FileCheck supports lookahead, I can write:

```
// CHECK-NOT: {{call (?!void @llvm\.assume)}}
```

But it doesn’t, so I have to use a more complex regular expression. The idea is:

```
   (?!d) =>                          [^d]|$
  (?!cd) =>   c(?!d)|[^c]|$ =>     c([^d]|$)|[^c]|$
 (?!bcd) =>  b(?!cd)|[^b]|$ =>   b(c([^d]|$)|[^c]|$)|[^b]|$
(?!abcd) => a(?!bcd)|[^a]|$ => a(b(c([^d]|$)|[^c]|$)|[^b]|$)|[^a]|$
```

So `(?!void @llvm\.assume)` becomes that complex regular expression.

Reply:
@scottmcm: The problem is that the test is compiled and tested under different configurations, and not all of them have the second `call void @llvm.assume`, adding `CHECK: call void @llvm.assume` will cause test failure on certain configurations.

Reply:
Using revisions to separate the targets could help, see `tests/assembly/simd-bitmask.rs` for an example.